### PR TITLE
Fixes #18170 - Session authenticator keeps asking for user

### PR DESCRIPTION
**Pull request by tstrachota**

### CODE REVIEW

File: test/unit/api/interactive_basic_auth_test.rb [hunk 1]
```diff
@@ -53,11 +53,10 @@
   end
 
   describe '#user' do
-    it "asks for username whe it wasn't provided" do
+    it "returns nil when it wasn't provided" do
```
Comment (mbacovsky) [inline test/unit/api/interactive_basic_auth_test.rb]: * returns nil when username wasn't provided

File: lib/hammer_cli_foreman/api/session_authenticator_wrapper.rb [hunk 2]
```diff
@@ -32,7 +32,8 @@ def status
       def authenticate(request, args)
         load_session
 
-        destroy_session if (@authenticator.user != @user)
+        user = @authenticator.user
+        destroy_session if (user && user != @user)
```
Comment (mbacovsky) [inline lib/hammer_cli_foreman/api/session_authenticator_wrapper.rb]: do we test somewhere that the session persist when user is not set?

**Comment by tstrachota** [comment]
@mbacovsky updated

**Comment by ares** [comment]
:+1: from me, fixes the issue and test seems alright

**Comment by tstrachota** [comment]
Since @mbacovsky is on PTO, his requests were implemented and @ares acked, I'm going to merge the PR.

**Pull request merged**